agent/connect/ca/provider_vault.go

```diff
@@ -227,6 +227,7 @@ func (v *VaultProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 	// Use the leaf cert role to sign a new cert for this CSR.
 	response, err := v.client.Logical().Write(v.config.IntermediatePKIPath+"sign/"+VaultCALeafCertRole, map[string]interface{}{
 		"csr": pemBuf.String(),
+		"ttl": fmt.Sprintf("%.0fh", v.config.LeafCertTTL.Hours()),
```

Comment:
This will round by truncation. If the user configured less than 1 hour won't this make TTL be `0h` which is not going to work well :)

Moot point if we enforce config can't be less than 1 somewhere but even so a comment would be good.